src/passes/I64ToI32Lowering.cpp

```diff
@@ -38,6 +38,48 @@ static Name makeHighName(Name n) {
 }
 
 struct I64ToI32Lowering : public WalkerPass<PostWalker<I64ToI32Lowering>> {
+  struct TempVar {
+    TempVar(Index idx, I64ToI32Lowering& pass) :
+      idx(idx), pass(pass), moved(false) {}
+
+    TempVar(TempVar&& other) : idx(other), pass(other.pass), moved(false) {
+      assert(!other.moved);
+      other.moved = true;
+    }
+
+    TempVar& operator=(TempVar&& rhs) {
+      assert(!rhs.moved);
```

Comment:
What happens if:
```c++
TempVar a(1, pass);
TempVar b(2, pass);
TempVar c(3, pass);
a = b;
b = c;
```
?
(having only looked at this struct in this PR)

Reply:
This code doesn't compile because the assignment operator has been deleted. However, if you used `std::move` to use move assignment, you're totally right that this is completely broken. It leaks 1 and 3 and frees 2. I'll fix this when I have time, probably tomorrow or Friday.

Reply:
Are you sure? This seems to work OK: https://godbolt.org/g/zFjpFw

Reply:
It would compile fine but it would leak the `1` when `a` was overwritten with `b`, since destroying `a` after that frees the `2` instead of the `1`. However, the sample code you linked contains the fixed version that frees the old value before overwriting it in the move assignment operator.

Reply:
A sorry, didn't realize I was looking at the fixed code. :-)